packages/api/src/utils/db-client.js

```diff
@@ -10,6 +11,165 @@ export class DBClient {
         Accept: '*/*',
       },
     })
+    this._clientNftStorage = new PostgrestClient(endpoint, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+        Accept: '*/*',
+      },
+      schema: 'nftstorage',
+    })
+  }
+
+  /**
+   * @param {Object} permaCache
+   * @param {number} permaCache.userId
+   * @param {string} permaCache.url
+   * @param {number} permaCache.size
+   * @param {string} permaCache.insertedAt
+   */
+  async createPermaCache(permaCache) {
+    const { data, error, status } = await this._client
+      .from('perma_cache')
+      .insert({
+        user_id: permaCache.userId,
+        url: permaCache.url,
+        size: permaCache.size,
+        inserted_at: permaCache.insertedAt,
+      })
+      .single()
+
+    if (error) {
+      if (status === HTTP_STATUS_CONFLICT) {
+        throw new ConstraintError({
+          message: 'URL already found for user',
+          status: HTTP_STATUS_CONFLICT,
+        })
+      }
+      throw new DBError(error)
+    }
+
+    if (!data) {
+      throw new Error('Perma cache not created.')
+    }
+
+    return data
+  }
+
+  /**
+   * Get perma cache Entry
+   *
+   * @param {number} userId
+   * @param {string} url
+   * @return {Promise<{ url: string, size: number }>}
+   */
+  async getPermaCache(userId, url) {
+    const { data, status, error } = await this._client
+      .from('perma_cache')
+      .select(
+        `
+      url,
+      size,
+      insertedAt:inserted_at
+      `
+      )
+      .eq('user_id', userId)
+      .eq('url', url)
+      .is('deleted_at', null)
+      .single()
+
+    if (status === 406 || !data) {
+      return
+    }
+    if (error) {
+      throw new DBError(error)
+    }
+
+    return data
+  }
+
+  /**
+   * List perma cache
+   *
+   * @param {number} userId
+   * @param {Object} [opts]
+   * @param {number} [opts.size=10]
+   * @param {string} [opts.before]
+   * @param {'Date'|'Size'} [opts.sortBy='inserted_at']
+   * @param {'Desc'|'Asc'} [opts.sortOrder='Desc']
+   * @return {Promise<Array<{ url: string, size: number, insertedAt: string }>>}
+   */
+  async listPermaCache(userId, opts = {}) {
+    let query = this._client
+      .from('perma_cache')
+      .select(
+        `
+      url,
+      size,
+      insertedAt:inserted_at
+      `
+      )
+      .eq('user_id', userId)
+      .is('deleted_at', null)
+      .limit(opts.size || 10)
+      .order(opts.sortBy === 'Size' ? 'size' : 'inserted_at', {
+        ascending: opts.sortOrder === 'Asc',
+      })
+
+    if (opts.before) {
+      query = query.lt('inserted_at', opts.before)
```

Comment:
This just, doesn't make sense when ordering by size...

Reply:
I will add support for sorting in next PR to not increase the scope here

Reply:
ok, I ended up doing it with suggested comment above